### PR TITLE
rados: free cluster runtime resources automatically

### PR DESCRIPTION
**Pull request by nixpanic**
Release resources that are allocated while configuring the connection to
the cluster. rados_shutdown() should only be needed after a successful
call to rados_connect(), however if the connection has been configured
with non-default parameters, some of the parameters may be allocated
before connecting. rados_shutdown() will free the allocated resources,
even if there has not been a connection yet.

Note that the finalizers get executed during garbage collection, which
can be forced by calling runtime.GC() for testing.

Fixes: #109

### CODE REVIEW

File: rados/conn.go [hunk 1]
```diff
@@ -59,6 +59,8 @@ func (c *Conn) Shutdown() {
 		return
 	}
 	C.rados_shutdown(c.cluster)
+	// set to nil, so that freeConn() does not call rados_shutdown() again.
+	c.cluster = nil
```
Comment (phlogistonjohn) [inline rados/conn.go]: This is a good change but I think we ought to make both the freeConn / Shutdown functions as idempotent as we can. It strikes me that both functions should behave similarly such that they share most of their behavior and/or code. (Call freeConn from within Shutdown?)
What do you think?

**Comment by phlogistonjohn** [comment]
To correct my comment above, the example code in question is posted by @nixpanic in the issue #109 - that's the code I was referring to that should be part of the repo somehow.

File: rados/conn.go (hunk 1 above)
Comment (nixpanic) [reply rados/conn.go]: Yes, that might be cleaner. I'll change it.

**Comment by nixpanic** [comment]
Please check again, the test has been added too.

**Pull request merged**